speedtest/test_performance_openssl.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/python
+"""
+A py.test test that attempts to build openssl and benchmark the effect of clcache
+"""
+import sys
+import os
+import shutil
+import pytest
+import urllib
+import zipfile
+import subprocess
+import time
+
+
+OPENSSL_ZIP = "OpenSSL_1_0_2-stable.zip"
+OPENSSL_URL = "https://codeload.github.com/openssl/openssl/zip/OpenSSL_1_0_2-stable"
+
+THISDIR = os.path.dirname(os.path.abspath(__file__))
+DISTDIR = os.path.join(os.path.dirname(THISDIR), "dist")
+CLCACHE = os.path.join(DISTDIR, "cl.exe")
+SOURCES = []
+ENVS = dict(os.environ)
+os.chdir(THISDIR)
+
+
+def retry_delete(path):
+    """
+    Repeatedly attempt to delete path
+    :param path:
+    :return:
+    """
+    for _ in range(30):
+        # antivirus might be busy in here..
+        try:
+            shutil.rmtree(path)
+            return
+        except WindowsError:
+            time.sleep(2)
+        if os.path.exists(path):
+            raise Exception("could not delete {}".format(path))
+
+
+class BlockMessage(object):
+    """
+    Little class to emit "begin .. end" messages for a block of code
+    """
+
+    def __init__(self, message):
+        self.started = 0
+        self.message = message
+
+    def __enter__(self):
+        self.started = time.time()
+        print "\n..begin {} .. ".format(self.message)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        result = "OK"
+        if exc_val is not None:
+            result = "ERROR"
+        print "\n..end {} {}.. ({}sec)".format(self.message, result,
+                                               time.time() - self.started)
+
```

Comment:
Using `contextlib.contextmanager`, this class could be simplified to the generator

```python
@contextmanager
def blockMessage(message):
    started = time.time()
    print "\n..begin {} .. ".format(message)
    try:
        yield
        result = "OK"
    except:
        result = "ERROR"
    print "\n..end {} {}.. ({}sec)".format(message, result, time.time() - started)
```